calender 구현

HAEROOL opened the pull request:
고칠점
- date 객체 redux로 넘기기

HAEROOL committed:
calender 구현

고칠점
- date 객체 redux로 넘기기

sangyun5108 committed:
making checkType

연도,월,일 입력시에 수가아닌 문자를 입력하는 경우 입력창을 현재 날짜로 초기화 해주는 기능 구현, 입력 날짜 및 개월수를 10보다 작은 수를 입력하는 경우 최종적으로 01,02,03,..과 같은 방식으로 input창에 보여지도록 구현

sangyun5108 committed:
Merge branch 'accountbook' of https://github.com/sangyun5108/BCSDproject into accountbook

SimYunSup commented on src/components/ShowCalender.js:
```diff
@@ -0,0 +1,92 @@
+import React from 'react'
+import styled from 'styled-components'
+function BuildCalender({date}){
+    const Dates = styled.div`
```
styled-components는 보통 컴포넌트 밖에서 스타일링합니다 :)
https://styled-components.com/docs/basics#define-styled-components-outside-of-the-render-method

SimYunSup commented on src/components/ShowCalender.js:
```diff
@@ -0,0 +1,92 @@
+import React from 'react'
+import styled from 'styled-components'
+function BuildCalender({date}){
```
컴포넌트의 이름은 보통 명사로 짓습니다. BuildCalendar는 컴포넌트의 이름보다는 함수의 이름에 더 걸맞네요.

SimYunSup commented on src/components/Calender.js:
```diff
@@ -0,0 +1,37 @@
+import React,{useState} from 'react'
+import ShowCalender from './ShowCalender'
+function ShowDate({date, setDate}){//날짜 출력
```
보통 한 파일당 컴포넌트 하나를 넣습니다.

SimYunSup commented on src/components/ShowCalender.js:
```diff
@@ -0,0 +1,92 @@
+            IsMonth : false
+        })
+        nextMonthday++
+    }//6째주 마저 채우기
```
너무 하드코딩이네요. 수요일까지 어떻게 코드를 효율적으로 짤까 고민해보세요.
팁을 드리자면 Array.from과 Date의 set메서드들을 잘 활용해주세요.

SimYunSup commented on src/components/ShowCalender.js:
```diff
@@ -0,0 +1,92 @@
+        </div>
+    )   
+}
+function ShowCalender({date}){
```
위의 이름 지적과 같습니다. 그리고 Calendar 컴포넌트와 같게 한 파일당 한 컴포넌트를 선언합니다.

SimYunSup commented on src/components/ShowCalender.js:
```diff
@@ -0,0 +1,92 @@
+        background : lightgray;
+        display : inline-block;
+        border : red 1px solid;
+        width : 13.855%;
```
flex box를 이용해주세요..

SimYunSup commented on src/components/ShowCalender.js:
```diff
@@ -0,0 +1,92 @@
+    const dayArray = ['Sun','Mon','Tue','Wed','Thu','Fri','Sat']
+    return (
+        <Container>
+            {dayArray.map((day,index) => <DOW>{day}</DOW>)}
```
header를 감싸는 엘리먼트는 없나요?

SimYunSup reviewed (changes requested):
어떻게 하면 코드를 간단하게 만들 수 있을지 고민하는게 우선일 것 같네요 :) 그리고 CSS Flex box에 대한 공부와 같이요. Grid Layout을 써도 상관은 없지만 CSS에 대한 공부를 더 했으면 합니다. 그리고 상윤님과 해루님 회의 전까지는 마음대로 merge하지 않으셨으면 합니다.

sangyun5108 committed:
localStorage

income과 expediture에 데이터를 입력할때마다, localStorage.setItem을 사용해서 데이터를 저장해주고, reducer의 initialState의 값을 만약 localStorage에 데이터가 있다면, 가져오고 아닌경우 빈 배열을 갖도록 구현했습니다

HAEROOL committed:
Calender 구현 수정 및 추가사항

수정사항
- 한 파일당 한 컴포넌트 선언으로 수정했습니다.
- Array.from을 사용하여 calender를 출력하는 함수를 수정했습니다.
- ShowCalender 파일을 삭제하고 MainCalender를 새로 만들었습니다.
- styled-components룰 컴포넌트 함수 바깥으로 뺐습니다.

추가 사항
- 달력에 localStorage에서 list를 받아와 소득/소비를 출력하는 기능 추가했습니다.
- 각 컴포넌트를 container로 묶을 수 있는 건 묶어서 정리했습니다.

sangyun5108 committed:
edit month data

기존에 문자를 이용하여 month를 사용했던 부분을 숫자로 수정했습니다.

HAEROOL committed:
Additional fix

- 상윤님 올리신 localStorage 데이터 기반으로 calender의 소득/소비 출력하는 기능 중에 문자열로 월을 받아오는 것을 숫자로 변경하였습니다

HAEROOL committed:
Calender fixed

Main Calender의 알고리즘을 변경하였습니다.

HAEROOL committed:
Calender incomce/expediture

account book에서 데이터를 가져와 calender에 income과 expediture를 출력하도록 했습니다.

HAEROOL committed:
Calender CSS

Calender CSS를 약간 수정했습니다.

HAEROOL committed:
Calender CSS

Calender 날짜를 출력하는 부분의 CSS를 추가하였습니다.

HAEROOL committed:
redux action, CSS

redux에 calender 금액 액션 추가 및 CSS 수정
